### PR TITLE
Move DirectDecoder to dwio::common

### PR DESCRIPTION
**Pull request by yingsu00**
DirectDecoder will be used by both DWRF and Parquet readers. This PR moves it and its parent class IntDecoder to dwio::common. The encoders were not moved.

Depends on https://github.com/facebookincubator/velox/pull/1849

### CODE REVIEW

**Comment by yingsu00** [comment]
@HuamengJiang will you be able to review this PR? It'll be very appreciated!

**Comment by weizheng10** [comment]
I am reviewing it. Expect to send the feedback out by EOD today.

From: Ying ***@***.***>
Date: Wednesday, June 29, 2022 at 9:34 PM
To: facebookincubator/velox ***@***.***>
Cc: Wei Zheng ***@***.***>, Comment ***@***.***>
Subject: Re: [facebookincubator/velox] Move DirectDecoder to dwio::common (PR #1865)

@HuamengJiang<https://github.com/HuamengJiang> will you be able to review this PR? It'll be very appreciated!

File: velox/dwio/common/DirectDecoder.h [hunk 1]
```diff
@@ -16,11 +16,11 @@
 
 #pragma once
 
+#include "velox/dwio/common/IntDecoder.h"
```
[the remark this answers is not in the document]
Comment (yingsu00) [reply velox/dwio/common/DirectDecoder.h:19]: Format check in both Circle CI and local format check does not like that order and would report an error. I don't know why. Similarly there is VectorFunction.cpp where local "make format-check" doesn't like the space after `-> auto {` and would make it `-> auto{`

File: velox/dwio/common/IntDecoder.cpp [hunk 2]
```diff
@@ -14,13 +14,11 @@
  * limitations under the License.
  */
 
-#include "velox/dwio/dwrf/common/IntDecoder.h"
+#include "velox/dwio/common/IntDecoder.h"
+#include "velox/dwio/common/DirectDecoder.h"
```
[the remark this answers is not in the document]
Comment (yingsu00) [reply velox/dwio/common/IntDecoder.cpp:18]: @weizheng10 I was informed we want full include path even for the corresponding .h file. Then, the includes should be sorted in alphabet order, and  the rules were already enforced by format check. I do see some cpp files just follow alphabet order , example:
ByteRLE.cpp
```
#include "velox/dwio/common/exception/Exceptions.h"
#include "velox/dwio/dwrf/common/ByteRLE.h"
```
Also, I saw some cpp files include doesn't have an empty line after the corresponding .h"
RLEv2.cpp
```
#include "velox/dwio/dwrf/common/RLEv2.h"
#include "velox/dwio/common/SeekableInputStream.h"
#include "velox/dwio/dwrf/common/Common.h"
```
PagedInputStream.cpp
```
#include "velox/dwio/dwrf/common/PagedInputStream.h"
#include "velox/dwio/common/SeekableInputStream.h"
#include "velox/dwio/common/exception/Exception.h"
```
while some do
compression.cpp
```
#include "velox/dwio/dwrf/common/Compression.h"

#include "velox/dwio/common/Common.h"
#include "velox/dwio/common/compression/LzoDecompressor.h"
#include "velox/dwio/common/exception/Exception.h"
#include "velox/dwio/dwrf/common/PagedInputStream.h"
#include "velox/dwio/dwrf/common/PagedOutputStream.h"

#include <folly/logging/xlog.h>
#include <lz4.h>
#include <snappy.h>
#include <zlib.h>
#include <zstd.h>
#include <zstd_errors.h>
```
Furthermore, the system includes are sometimes in front sometimes after the velox includes.
ColumnSelector.cpp
```
#include "velox/dwio/common/ColumnSelector.h"

#include <iterator>

#include "folly/Conv.h"
#include "velox/dwio/common/TypeUtils.h"
```
BufferedInput.cpp
```
#include <fmt/format.h>

#include "velox/dwio/common/BufferedInput.h"
```
Do we have a standard for the includes? cc @pedroerp @mbasmanova

[the remark this answers is not in the document]
Comment (yingsu00) [reply velox/dwio/common/IntDecoder.cpp:18]: One more thing, I saw most velox includes were using double quotation marks, but some are using `<>`
HashPartitionFunction.cpp
```
#include <velox/exec/HashPartitionFunction.h>
#include <velox/exec/VectorHasher.h>
```
Should this be changed?

**Comment by yingsu00** [comment]
@weizheng10 Thank you very much for reviewing. I have addressed your comments and would appreciate your review again.

**Comment by yingsu00** [comment]
@weizheng10 @zzhao0 Will you be able to review this PR again? Many thanks!

File: velox/dwio/common/IntDecoder.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (HuamengJiang) [reply velox/dwio/common/IntDecoder.cpp]: > 



> One more thing, I saw most velox includes were using double quotation marks, but some are using `<>` HashPartitionFunction.cpp
> 
> ```
> #include <velox/exec/HashPartitionFunction.h>
> #include <velox/exec/VectorHasher.h>
> ```
> 
> Should this be changed?

I noticed that sometimes VS code does this when it automatically includes files for you. My understanding is that the convention we prefer is still to use double quotes.

**Comment by HuamengJiang** [comment]
Sorry for being late to the party. 😂 I didn't get notifications on this for some reason.

The changes look reasonable by themselves. I'll need to take another pass for nits. However, I noticed that @oerling imported this into fbcode to fix the build there but the diff is work in progress. I'm following up with him there to figure out a couple of build breaks. Some changes likely need to back propagated to the PR.

**Comment by weizheng10** [comment]
Yes, please use “”.

From: RindsSchei225e ***@***.***>
Date: Thursday, July 7, 2022 at 12:01 AM
To: facebookincubator/velox ***@***.***>
Cc: Wei Zheng ***@***.***>, Mention ***@***.***>
Subject: Re: [facebookincubator/velox] Move DirectDecoder to dwio::common (PR #1865)

@HuamengJiang commented on this pull request.

________________________________

In velox/dwio/common/IntDecoder.cpp<https://github.com/facebookincubator/velox/pull/1865#discussion_r915526308>:

> @@ -14,13 +14,11 @@

  * limitations under the License.

  */



-#include "velox/dwio/dwrf/common/IntDecoder.h"

+#include "velox/dwio/common/IntDecoder.h"

+#include "velox/dwio/common/DirectDecoder.h"

One more thing, I saw most velox includes were using double quotation marks, but some are using <> HashPartitionFunction.cpp

#include <velox/exec/HashPartitionFunction.h>

#include <velox/exec/VectorHasher.h>

Should this be changed?

I noticed that sometimes VS code does this when it automatically includes files for you. My understanding is that the convention we prefer is still to use double quotes.

File: velox/dwio/common/IntDecoder.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (weizheng10) [reply velox/dwio/common/IntDecoder.cpp]: yes, please change them.

**Comment by yingsu00** [comment]
@HuamengJiang @weizheng10 Thanks for reviewing. Orri is out of office until next Monday. How much change is it there in the internal diff to resolve the conflicts? Can I help in anyway before he comes back?

**Comment by HuamengJiang** [comment]
> @HuamengJiang @weizheng10 Thanks for reviewing. Orri is out of office until next Monday. How much change is it there in the internal diff to resolve the conflicts? Can I help in anyway before he comes back?

I'm rebasing the imported diff and figuring out right now. Will let you know by EoD if there's significant changes. Otherwise I will just clean it up after merging.

**Comment by HuamengJiang** [comment]
@yingsu00, I have fixed all the builds locally. I'll proceed to merge and land.

**Comment by yingsu00** [comment]
Thank you so much @HuamengJiang !

**Comment by oerling** [comment]
All includes in Velox are “Velox/…/…h”

 

All includes of non-Velox content are <>.

**Pull request closed without merging**